Makefile

```diff
@@ -60,6 +60,9 @@ clean:
 
 build/llvm.BUILT:
 	mkdir -p build/llvm
+	# Apply https://reviews.llvm.org/D153293 as a patch until we're ready to
+	# upgrade LLVM (presumably once v17 has been released)
+	(cd $(LLVM_PROJ_DIR) && patch -p1 -N < ../../llvm-D153293-backport.patch) || true
```

Comment:
The annoying thing about this is it leaves the llvm-project submodule in a "dirty" state. I'm not sure if there's a way to reliably avoid that besides copying the whole source tree to a temporary directory and patching that instead, which feels like overkill.

Reply:
Of course it won't be necessary at all once we upgrade to LLVM 17.

Reply:
You should be able rebase to get LLVM 17.0.1; I guess the patch file can go away?